hail/python/hail/ir/ir.py

```diff
@@ -602,6 +602,35 @@ def _compute_type(self, env, agg_env):
         self._type = tndarray(self.nd.typ.element_type, self.nd.typ.ndim - len(self.axes))
 
 
+class NDArrayMatMul(IR):
+    @typecheck_method(l=IR, r=IR)
+    def __init__(self, l, r):
+        super().__init__(l, r)
+        self.l = l
+        self.r = r
+
+    @typecheck_method(l=IR, r=IR)
+    def copy(self, l, r):
+        return NDArrayMatMul(l, r)
+
+    def _compute_type(self, env, agg_env):
+        self.l._compute_type(env, agg_env)
+        self.r._compute_type(env, agg_env)
+        l_ndim = self.l.typ.ndim
+        r_ndim = self.r.typ.ndim
+
+        if l_ndim == 1 and r_ndim == 1:
+            ndim = 0
+        elif l_ndim == 1 or r_ndim == 1:
+            ndim = 1
```

Comment:
is this case right? It disagrees with the above, no?

Reply:
Ah ya, I think that got caught up in a rebase

Reply:
Gonna extract it anyway so it'll use the other one.